lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -2100,10 +2116,11 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
     assert(op.getNumResults() == 1 && "firrtl only has single-op results");
     Value result = op.getResult(0);
 
-    // If this is an expression emitted inline or unused, it doesn't need a
-    // name.
     bool isExpr = isVerilogExpression(&op);
-    if (isExpr) {
+    bool isModport = isa<sv::GetModportOp>(op);
```

Comment:
I would recommend using a dynamic cast, then using the cast'ed variable below in `addName`.